src/Tmds.Systemd/ServiceManager.cs

```diff
@@ -33,5 +37,57 @@ public static string InvocationId
                 return _invocationId;
             }
         }
+
+        private static bool CheckServiceManager()
+        {
+            // No point in testing anything unless it's Unix
+            if (Environment.OSVersion.Platform != PlatformID.Unix)
+            {
+                return false;
+            }
+
+            // We've got invocation id, it's systemd >= 232 running a unit
+            if (InvocationId != null)
+            {
+                return true;
+            }
+
+            // Either it's not a service, or systemd is < 232, do a bit more digging
+            try
+            {
+                // Test parent process
+                var parentPid = getppid();
+                var pidString = parentPid.ToString(NumberFormatInfo.InvariantInfo);
+
+                // If parent PID is not 1, this may be a user unit, in this case it must match MANAGERPID envvar
+                if (parentPid != 1
+                    && Environment.GetEnvironmentVariable("MANAGERPID") != pidString)
```

Comment:
wdyt about simplifying this method to checking MANAGERPID envvar isnt null or empty?

Reply:
ah, we still need the pid check. We could leave out the commFile part below.

return getppid() == 1 || !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MANAGERPID"));

Reply:
> getppid() == 1

Hmm, in a container your parent may be 1 but not systemd. ok, keep the commFile part.

Reply:
Yes, the pid == 1 + commFile check is for the case that systemd isn't used as a pid-1 process.
As fot the `MANAGERPID` presence, my first test was exactly for string.IsNullOrEmpty, but I thought that stricter check is fine in this case, both logic-wise (the spec says that MANAGERPID is present there) and performance-wise (in most common, positive case, we still have to allocate pidString).